apps/minifront/src/state/dutch-auction/constants.ts

```diff
@@ -0,0 +1,26 @@
+export const DURATION_OPTIONS = ['10min', '30min', '1h', '2h', '6h', '12h', '24h', '48h'] as const;
+export type DurationOption = (typeof DURATION_OPTIONS)[number];
+
+/**
+ * Blocks are ~5 seconds long (so, 12 blocks/minute), and the minimum duration
+ * for an auction is 10 minutes (so, 120 blocks). All other auction durations
+ * are even multiples of 10 minutes. So we'll set the step count to 120, so that
+ * the sub-auctions can divide evenly into the number of intervening blocks.
+ */
+export const STEP_COUNT = 120n;
+
+const APPROX_BLOCK_DURATION_MS = 5_000n;
+const MINUTE_MS = 60_000n;
+export const BLOCKS_PER_MINUTE = MINUTE_MS / APPROX_BLOCK_DURATION_MS;
+const BLOCKS_PER_HOUR = BLOCKS_PER_MINUTE * 60n;
+
+export const DURATION_IN_BLOCKS = new Map<DurationOption, bigint>([
+  ['10min', 10n * BLOCKS_PER_MINUTE],
+  ['30min', 30n * BLOCKS_PER_MINUTE],
+  ['1h', BLOCKS_PER_HOUR],
+  ['2h', 2n * BLOCKS_PER_HOUR],
+  ['6h', 6n * BLOCKS_PER_HOUR],
+  ['12h', 12n * BLOCKS_PER_HOUR],
+  ['24h', 24n * BLOCKS_PER_HOUR],
+  ['48h', 48n * BLOCKS_PER_HOUR],
+]);
```

Comment:
Ah, good call. I think this specific implementation is a little confusing, though, since it's not immediately clear what the numbers in the map represent. (In fact, in your example, you multiplied the `STEP_COUNT` by `10n` even though `STEP_COUNT` already represented ten minutes.)

However, I think _just_ adding tests should prevent the issue you described from happening in the future. I wrote [a test suite](https://github.com/penumbra-zone/web/pull/967/commits/16577516105060a61957dbd9c60e21c5a9a72d36) to ensure that durations are cleanly divisible by step counts.